README.md

```diff
@@ -1,4 +1,134 @@
-# Trampoline V2
+# docker-ci-helper
 
-This is a shell script for running another script in a Docker
-container.
+This repository contains `trampoline_v2.sh`, a shell script for
+running another script in a Docker container. Primary use case of this
+shell script is to use it as a test driver on various CI systems. By
+utilizing Docker, your tests will be very portable so that you can run
+them on any CI systems as well as on your workstations with minimum
+efforts.
+
+## trampoline_v2.sh
+
+This script does 3 things.
+
+1. Prepare the Docker image for the test.
+2. Run Docker with appropriate flags to run the test.
+3. Upload the newly built Docker image.
+
+We'll discuss each steps in detail.
+
+### Prepare the Docker image
+
+1. Try to download a Docker image specified by `TRAMPOLINE_IMAGE`
```

Comment:
I assume these are mutually exclusive. Does one have higher priority?

Reply:
It download the image

then
if `TRAMPOLINE_DOCKERFILE` is specified, it always build the image, utilizing the downloaded image as a cache.

Both can happen.